views/js/runner/provider/qti.js

```diff
@@ -89,7 +89,36 @@ define([
                 'serviceController',
                 'serviceExtension'
             ]);
-            return proxyFactory('qtiServiceProxy', proxyConfig);
+            return proxyFactory('qtiServiceProxy', proxyConfig)
+                // middle ware invoked on every requests
+                .use(function (req, res, next) {
+                    var context = self.getTestContext();
+                    var data = res && res.data;
+
+                    // test has been closed/suspended => redirect to the index page after message acknowledge
+                    if (data && data.type && data.type === 'TestState') {
+
+                        if(!self.getState('ready')){
+                            //if we open an inconsistent test (should never happen) we let a few sec to
+                            //read the error message then leave
+                            _.delay(function (){
```

Comment:
This part was taken from the previous implementation in the controller. According to the comment above, this delay was added to let the user see the error message.
But I agree it is not very useful as the browser will redirect the page anyway.
So I can remove it if you prefer.